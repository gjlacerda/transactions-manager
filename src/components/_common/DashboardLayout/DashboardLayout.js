import React from 'react'
import Header from 'components/Header'
import styled from 'styled-components'
import { marginSize } from 'utils/margin'

const Content = styled.div`
  max-width: 600px;
  margin: ${marginSize.large} auto;
`

export default ({ children }) => (
  <React.Fragment>
    <Header />
    <Content>
      {children}
    </Content>
  </React.Fragment>
)
