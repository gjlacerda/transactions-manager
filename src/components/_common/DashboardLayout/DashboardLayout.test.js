import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { BrowserRouter } from 'react-router-dom'
import DashboardLayout from './DashboardLayout'

describe('DashboardLayout component', () => {
  it('deve ser igual ao snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <DashboardLayout>
          <span>teste</span>
        </DashboardLayout>
      </BrowserRouter>,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
