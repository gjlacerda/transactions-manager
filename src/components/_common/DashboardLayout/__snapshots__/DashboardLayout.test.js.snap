// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardLayout component deve ser igual ao snapshot 1`] = `
Array [
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 20px;
  background-color: #1976d2;
  height: 50px;
}

.c1 {
  color: #ffffff;
  width: 25%;
  font-weight: 700;
  font-size: 20px;
}

.c2 {
  display: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50%;
}

.c3 {
  opacity: 1;
  -webkit-transition: opacity 0.1s ease-in-out;
  transition: opacity 0.1s ease-in-out;
  padding: 15px 0;
  color: #ffffff;
  margin: 0 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 3px solid #ffffff;
}

.c3:hover {
  opacity: 1;
}

.c4 {
  opacity: 0.8;
  -webkit-transition: opacity 0.1s ease-in-out;
  transition: opacity 0.1s ease-in-out;
  padding: 15px 0;
  color: #ffffff;
  margin: 0 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 3px solid transparent;
}

.c4:hover {
  opacity: 1;
}

@media (min-width:600px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      TM.
    </div>
    <nav
      className="c2"
    >
      <a
        active="true"
        className="c3"
        href="/"
        onClick={[Function]}
      >
        transações
      </a>
      <a
        active=""
        className="c4"
        href="/transaction"
        onClick={[Function]}
      >
        nova transação
      </a>
    </nav>
    <div />
  </div>,
  .c0 {
  max-width: 600px;
  margin: 40px auto;
}

<div
    className="c0"
  >
    <span>
      teste
    </span>
  </div>,
]
`;
